go/types/meta_sequence.go

```diff
@@ -339,3 +339,31 @@ func (es emptySequence) getCompositeChildSequence(start uint64, length uint64, h
 func (es emptySequence) isLeaf() bool {
 	return es.height == 1
 }
+
+// Invokes |cb| on all "uncommitted" ptree nodes reachable from |v| in
+// "bottom-up" order. It traverses across values, but will only traverse out of
+// the chunk containing |v| in order to follow uncommitted ptree nodes.
+func iterateUncommittedChildren(v Value, cb func(sv Value)) {
+	if _, ok := v.(*Type); ok {
+		// Types can be in-memory cyclic, but they cannot reference uncommitted
+		// nodes. Just avoid traversing into them.
+		return
+	}
+
+	if c, ok := v.(Collection); ok {
+		if ms, ok := c.sequence().(metaSequence); ok {
+			// Only traverse uncommitted children of meta sequences
+			for _, mt := range ms.tuples {
+				if mt.child != nil {
+					iterateUncommittedChildren(mt.child, cb)
+					cb(mt.child)
+				}
+			}
+			return
+		}
+	}
+
+	v.WalkValues(func(v Value) {
+		iterateUncommittedChildren(v, cb)
+	})
```

Comment:
Yeah. I didn't at first and it didn't work. The issue is that You can nest collections that have uncommitted values within other values (i.e. a struct) and you need to find those. So you need to recursively walk the value, but only "leave" the current chunk when traversing an uncommitted ref. I'll add a comment.